File: client-app/src/app/models/faculty.ts
import { KnowledgeBranch } from "./knowledgeBranch";
import { Photo } from "./photo";

export interface Faculty {
	id: string;
	universityId: string;
	name: string;
	info: string;
	studentsCount: number;
	facultyPhoto: Photo;
	knowledgeBranches: KnowledgeBranch[];
}

export class FacultyFormValues {
	id: string = '';
	name: string = '';
	info: string = '';
	facultyPhoto?: Photo;
	knowledgeBranches?: KnowledgeBranch[];
}

File: Domain/Faculty.cs
namespace Domain
{
	public class Faculty
	{
		public Guid Id { get; set; }
		public string Name { get; set; }
		public string Info { get; set; }
		public int StudentsCount { get; set; }
		public Photo FacultyPhoto { get; set; }
		public HigherEducationFacility HigherEducationFacility { get; set; }
		public ICollection<KnowledgeBranch> KnowledgeBranches { get; set; } = new List<KnowledgeBranch>();
		public ICollection<Specialty> Specialties { get; set; } = new List<Specialty>();
	}
}